lightning/src/ln/msgs.rs

```diff
@@ -1515,6 +1599,186 @@ impl_writeable_len_match!(NodeAnnouncement, {
 	contents
 });
 
+impl Readable for QueryShortChannelIds {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let chain_hash: BlockHash = Readable::read(r)?;
+
+		// We expect the encoding_len to always includes the 1-byte
+		// encoding_type and that short_channel_ids are 8-bytes each
+		let encoding_len: u16 = Readable::read(r)?;
+		if encoding_len == 0 || (encoding_len - 1) % 8 != 0 {
+			return Err(DecodeError::InvalidValue);
+		}
+
+		// Must be encoding_type=0 uncompressed serialization. We do not
+		// support encoding_type=1 zlib serialization.
+		let encoding_type: u8 = Readable::read(r)?;
+		if encoding_type != EncodingType::Uncompressed as u8 {
+			return Err(DecodeError::InvalidValue);
+		}
+
+		// Read short_channel_ids (8-bytes each), for the encoding_len
+		// less the 1-byte encoding_type
+		let mut short_channel_ids = vec![];
```

Comment:
This needs a `Vec::with_capacity(cmp::max(ln::peer_channel_encryptor::LN_MAX_MSG_LEN / 8, (encoding_len - 1)/8))`

Reply:
Ah good call on allocations!  

Do you think we should move the upper-bound range check up to where we validate `encoding_length`? I'm thinking that if we get an `encoding_length` that causes an `LN_MAX_MSG_LEN` overflow we're receiving a malformed message and should reject it outright instead of limiting the allocation length.

Reply:
That'd be totally fine too, yea.

Reply:
`encoding_len` is a `u16` so the range check or limiting of allocation against `LN_MAX_MSG_LEN` seems superfluous.  You ok with just doing: `Vec::with_capacity(((encoding_len - 1) / 8) as usize);`?

Reply:
Oh! Right indeed. Sure, that works too, maybe with a comment noting its a u16.